bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/internal/PersistentInbox.java

```diff
@@ -183,6 +184,9 @@ protected void deactivate() {
         if (results.isEmpty()) {
             throw new IllegalArgumentException("No Thing with UID " + thingUID.getAsString() + " in inbox");
         }
+        if (newThingId != null && newThingId.split(AbstractUID.SEPARATOR).length > 1) {
```

Comment:
`newThingId.contains(SEPARATOR)` would be clearer IMO.

Reply:
Yes, I agree

Reply:
Changed